policies/policy_service_test.go

```diff
@@ -1297,6 +1297,69 @@ func testSortPolicies(t *testing.T, pm policies.PageMetadata, ags []policies.Pol
 	}
 }
 
+func TestDuplicatePolicy(t *testing.T) {
+	users := flmocks.NewAuthService(map[string]string{token: email})
+	svc := newService(users)
+
+	policySingleCopy := createPolicy(t, svc, "policy-single")
+
+	policyMultipleCopy := createPolicy(t, svc, "policy")
+	copy := createPolicy(t, svc, "policy_copy")
+	for i := 2; i <= 10; i++ {
+		name := fmt.Sprintf("policy_copy_%d", i)
+		_ = createPolicy(t, svc, name)
+	}
+
+	cases := map[string]struct {
+		policyID     string
+		expectedName string
+		token        string
+		err          error
+	}{
+		"duplicate a existing policy": {
+			policyID: policySingleCopy.ID,
+			expectedName: "policy-single_copy",
+			token:    token,
+			err:      nil,
+		},
+		"duplicate a existing policy with an invalid token": {
+			policyID: policySingleCopy.ID,
+			token:    invalidToken,
+			err:      policies.ErrUnauthorizedAccess,
+		},
+		"duplicate a existing policy with empty ID": {
+			policyID: "",
+			token:    token,
+			err:      policies.ErrNotFound,
+		},
+		"duplicate two times a existing policy": {
+			policyID: policyMultipleCopy.ID,
+			expectedName: "policy_copy_11",
+			token:    token,
+			err:      nil,
+		},
+		"duplicate a copy of a policy": {
+			policyID: copy.ID,
+			expectedName: "policy_copy_copy",
+			token:    token,
+			err:      nil,
+		},
+	}
+
+	for desc, tc := range cases {
+		t.Run(desc, func(t *testing.T) {
+			policyDuplicated, err := svc.DuplicatePolicy(context.Background(), tc.token, tc.policyID)
+			assert.True(t, errors.Contains(err, tc.err), fmt.Sprintf("%s: expected %s got %s", desc, err, tc.err))
+
+			originalPolicy, _ := svc.ViewPolicyByID(context.Background(), token, tc.policyID)
+			if err == nil {
+				assert.Equal(t, originalPolicy.Policy, policyDuplicated.Policy, fmt.Sprintf("%s: expected %v got %v", desc, originalPolicy, policyDuplicated))
+				assert.Equal(t, tc.expectedName, policyDuplicated.Name.String(),  fmt.Sprintf("%s: expected %v got %v", desc, tc.expectedName, policyDuplicated.Name))
+			}
```

Comment:
we only validate name here, we could add the most important attrs here to validate the duplicity